slkspec/core.py

```diff
@@ -152,6 +155,7 @@ def _retrieve_items(self, retrieve_files: list[tuple[str, str]]) -> None:
                 (output_dir / out_file.name).chmod(self.file_permissions)
 
     def _cache_files(self) -> None:
+        time.sleep(self.delay)
```

Comment:
Here is where potential race conditions might happen - try moving the sleep method into the thread lock block. But that might end you up with the same issue as before.